apps/workers/assetPreprocessingWorker.ts

```diff
@@ -67,24 +75,75 @@ async function readImageText(buffer: Buffer) {
 
 async function readPDFText(buffer: Buffer): Promise<{
   text: string;
-  metadata: Record<string, string>;
+  metadata: Record<string, object>;
 }> {
   return new Promise((resolve, reject) => {
-    // Need raw text flag represents as number (1), reference : https://github.com/modesty/pdf2json/issues/76#issuecomment-236569265
-    const pdfParser = new PDFParser(null, 1);
+    const pdfParser = new PDFParser(null, true);
     pdfParser.on("pdfParser_dataError", reject);
     pdfParser.on("pdfParser_dataReady", (pdfData) => {
       resolve({
-        // The type isn't set correctly, reference : https://github.com/modesty/pdf2json/issues/327
-        // eslint-disable-next-line
-        text: (pdfParser as any).getRawTextContent(),
+        text: pdfParser.getRawTextContent(),
         metadata: pdfData.Meta,
       });
     });
     pdfParser.parseBuffer(buffer);
   });
 }
 
+export async function extractAndSavePDFScreenshot(
+  pdfBuffer: Buffer,
+  userId: string,
+  bookmarkId: string,
+  jobId: string,
+): Promise<boolean> {
+  try {
+    console.log("extracting =================================");
+    logger.info(
+      `[${jobId}] Attempting to generate PDF screenshot for bookmarkId: ${bookmarkId}`,
+    );
+    /**
+     * If you encountered any issues with this library, make sure you have ghostscript and graphicsmagick installed following this URL
+     * https://github.com/yakovmeister/pdf2image/blob/HEAD/docs/gm-installation.md
+     */
+    const screenshot = await fromBuffer(pdfBuffer, {
+      density: 100,
+      quality: 100,
+      format: "png",
+      preserveAspectRatio: true,
+    })(1, { responseType: "buffer" });
+
+    if (!screenshot.buffer) {
+      logger.error(`[${jobId}] Failed to generate PDF screenshot`);
+      return false;
+    }
+
+    // Store the screenshot
+    const asset = await storeScreenshot(screenshot.buffer, userId, jobId);
+
+    if (!asset) {
+      logger.error(`[${jobId}] Failed to store PDF screenshot`);
+      return false;
+    }
+
+    // Insert into database
+    await db.insert(assets).values({
+      id: asset.assetId,
+      bookmarkId,
+      userId,
+      assetType: AssetTypes.LINK_SCREENSHOT,
```

Comment:
`LINK_SCREENSHOT` are meant to be in particular for "link" screenshots. This is more like `ASSET_SCREENSHOT`.